### PR TITLE
Dev15 early steps: build conditionals and dev14 artifacts folder

### PR DESCRIPTION
**Pull request by drewgillies**
This is the first of a number of PRs required to light up building against the Dev15 libraries.
Actual Dev15 building is still a few steps away. The $(VisualStudioVersion) == 15.0 is analogous to a feature flag.

### CODE REVIEW

**Comment by drewgillies** [comment]
Can you please take a look at this, @alpaix or @emgarten?

File: build/common.ps1 [hunk 1]
```diff
@@ -532,7 +533,7 @@ Function Invoke-ILMerge {
     Trace-Log 'Creating the ilmerged nuget.exe'
     $opts = , 'NuGet.exe'
     $opts += $buildArtifacts
-    $opts += "/out:$Artifacts\NuGet.exe"
+    $opts += "/out:$NuGetExeOutDev14\NuGet.exe"
```
Comment (alpaix) [inline build/common.ps1]: Why? The whole build pipeline relies on that location :confused:

Comment (drewgillies) [inline build/common.ps1]: Great catch. Thanks--what I'll do is remove the changes to the build script for now. When we come to build both Dev14 and Dev15 targets (which isn't yet), I'll make sure the Dev14 nuget.exe + pdb stay in the artifacts root. The VSIX will need to be stored in separate output dirs though, but that's taken care of in VsExtension.csproj.

**Comment by alpaix** [comment]
Just a thought. Is it possible to add a new solution configuration, say "Dev15 Debug", in order to bind all custom properties and build steps to it?

**Comment by drewgillies** [comment]
@alpaix It could definitely be done, but I don't know if it would have the desired effect. We'd need to support a Dev15 debug and Dev15 release configuration presumably, which would introduce more target noise than what we currently have in this PR.

File: build/common.ps1 (hunk 1 above)
Comment (drewgillies) [inline build/common.ps1]: Made this and other changes (mainly propagated csproj changes to test projects). Please review in following commit. Will squash these two.

**Comment by emgarten** [comment]
Can the floating System.\* package versions be changed to a static version?

**Comment by drewgillies** [comment]
@emgarten You mean by locking into a specific beta?

**Comment by drewgillies** [comment]
@emgarten [discussed offline] Have added a commit (will squash all three) which removes these from the project.dev15.json files. Have also removed them from the csproj files--they were unnecessarily referenced (prior to this PR).

**Comment by drewgillies** [comment]
Discussed further with @emgarten offline/email. Good to go. Commiting.

**Pull request closed without merging**